packages/workspace/src/browser/workspace-service.ts

```diff
@@ -57,18 +67,149 @@ export class WorkspaceService implements FrontendApplicationContribution {
 
     @postConstruct()
     protected async init(): Promise<void> {
-        const rootUri = await this.server.getRoot();
-        this._root = await this.toValidRoot(rootUri);
+        this._workspaceDescription = {
+            root: undefined,
+            roots: [],
+            workspaceConfigFile: undefined
+        };
+        await this.setupRoots();
+        this.updateTitle();
+        this.deferredWorkspaceDescription.resolve(this._workspaceDescription);
+        const configUri = await this.getWorkspaceConfigFileUri();
+        if (configUri) {
+            this.watcher.onFilesChanged(async event => {
+                const isRootConfigAddedOrChanged = !!event.find(change =>
+                    (change.type === FileChangeType.ADDED || change.type === FileChangeType.UPDATED) && change.uri.toString() === configUri.toString()
+                );
+                const isRootConfigRemoved = !!event.find(change =>
+                    change.type === FileChangeType.DELETED && change.uri.toString() === configUri.toString()
+                );
+                if (isRootConfigAddedOrChanged) {
+                    await this.onRootConfigFileChanged();
+                } else if (isRootConfigRemoved) {
+                    this.stopWatch();
+                    this.startWatch(this._root);
+                    if (this._root) {
+                        this._workspaceDescription.roots = [this._root];
+                    }
+                    this._workspaceDescription.workspaceConfigFile = undefined;
+                    this.onWorkspaceChangeEmitter.fire(this._workspaceDescription);
+                }
+            });
+        }
+
+        await this.openWindow({ preserveWindow: true });
```

Comment:
Why is it called here? to fire a workspace changed event? It should not do it, an event should be only fired on a workspace update in one place